client/allocrunner/taskrunner/task_runner.go

```diff
@@ -982,6 +982,10 @@ func (tr *TaskRunner) handleKill(resultCh <-chan *drivers.ExitResult) *drivers.E
 	if delay := tr.Task().ShutdownDelay; delay != 0 {
 		tr.logger.Debug("waiting before killing task", "shutdown_delay", delay)
 
+		ev := structs.NewTaskEvent(structs.TaskWaitingShuttingDownDelay).
+			SetDisplayMessage("Waiting for shutdown delay before killing the task.")
```

Comment:
For the display message we usually try to provide a little more information than is in the event type name. What if we told the user how long they'll be waiting:

```suggestion
			SetDisplayMessage(
				fmt.Sprintf("Waiting for shutdown_delay of %s before killing the task", delay))			
```